projects/packages/forms/src/blocks/field-file/edit.js

```diff
@@ -0,0 +1,99 @@
+import { getJetpackExtensionAvailability } from '@automattic/jetpack-shared-extension-utils';
+import {
+	useBlockProps,
+	useInnerBlocksProps,
+	store as blockEditorStore,
+} from '@wordpress/block-editor';
+import { useSelect } from '@wordpress/data';
+import { useMemo } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import clsx from 'clsx';
+import JetpackFieldControls from '../shared/components/jetpack-field-controls';
+import { UpsellNudge } from '../shared/components/upsell-nudge';
+import useFormWrapper from '../shared/hooks/use-form-wrapper';
+import useJetpackFieldStyles from '../shared/hooks/use-jetpack-field-styles';
+import useParentFormClientId from '../shared/hooks/use-parent-form-client-id';
+import './editor.scss';
+
+export default function FileFieldEdit( props ) {
+	const { attributes, clientId, isSelected, setAttributes, name, className } = props;
+	const { id, required, width } = attributes;
+
+	const fieldFileAvailability = getJetpackExtensionAvailability( 'field-file' );
+
+	useFormWrapper( { attributes, clientId, name } );
+	const { blockStyle } = useJetpackFieldStyles( attributes );
+
+	const classes = clsx( className, 'jetpack-field is-non-animated-label', {
+		'is-selected': isSelected,
+		[ `jetpack-field__width-${ width }` ]: width,
+	} );
+
+	const blockProps = useBlockProps( {
+		className: classes,
+		style: blockStyle,
+	} );
+
+	const uploadLabel = __( 'Upload a file', 'jetpack-forms' );
+	const innerBlocksProps = useInnerBlocksProps( blockProps, {
+		template: [
+			[
+				'jetpack/label',
+				{
+					label: uploadLabel,
+					defaultLabel: uploadLabel,
+					lock: { move: true, remove: true },
+				},
+			],
+			[
+				'jetpack/dropzone',
+				{
+					lock: { move: true, remove: true },
+					layout: { type: 'flex', justifyContent: 'center', orientation: 'vertical' },
+				},
+			],
+		],
+		allowedBlocks: [ 'jetpack/dropzone', 'jetpack/label' ],
+		renderAppender: false,
+	} );
```

Comment:
I think this might need `templateLock: 'all',`, otherwise it's possible to duplicate the `dropzone` and wind up with two within the parent file field block:

<img width="977" alt="image" src="https://github.com/user-attachments/assets/fa65d489-6ea5-41b5-92ad-0b9b19a17a23" />

Reply:
I tried this but it make it so that I can't add any new blocks in the dropdown block. Which is not the desired behaviour. 
I managed to resolve the issue by making the allow list empty.